internal/tracing/conn_unix.go

```diff
@@ -0,0 +1,26 @@
+package tracing
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strings"
+	"syscall"
+)
+
+const maxUnixSocketPathSize = len(syscall.RawSockaddrUnix{}.Path)
+
+func DialInMemory(ctx context.Context, addr string) (net.Conn, error) {
```

Comment:
@thaJeztah I was thinking of PRing this to `docker/go-connections` later. Thoughts? It's inspired by the `InmemSocket` there

Reply:
@milas hmm. need to think about this; honest answer;

- there's an issue with the latest (main) version of `go-connections` that we still need to sort out (moby started failing when we tried to update it to latest)
- there's been some discussion about that repository (mainly around the TLS config specified in that repository, but also because it became a bit of a grab-bag of utilities that are not very related), and we need to re-think if we want to preserve it.


I wonder if we should have something like https://github.com/docker/go-metrics (either to provide abstractions, or to have a shared implementation for some of this).

(all just quick blurbs from my side! haven't dug in deep into all of this, but perhaps something we should discuss with more people)

Reply:
Ahhh sorry, I should have clarified -- I was only referring to the `DialInMemory` function - the way that library is structured right now, you can:
 * create a listener type that works across OSes (impls behind build tags)
 * configure an HTTP transport across OSes
But it doesn't expose a function to create a platform-agnostic in-memory `net.Conn` directly


(That said, yes -- the rest of the observability pieces should be shared! I purposefully put this in `internal/tracing` so it _cannot_ become a dependency elsewhere & it doesn't depend on anything in the rest of the Compose code base. Will spin it out into its own repo/module later)